src/Microsoft.VisualStudio.Services.Agent/JobServerQueue.cs

```diff
@@ -438,42 +356,27 @@ public void QueueTimelineRecordUpdate(Guid timelineId, TimelineRecord timelineRe
         {
             while (!_jobCompletionSource.Task.IsCompleted || runOnce)
             {
-                if (!runOnce)
-                {
-                    bool forceProcess = await _timelineUpdateQueueSemaphore.WaitAsync(_delayForTimelineUpdateDequeue);
-                    if (forceProcess)
-                    {
-                        Trace.Verbose("Process timeline update queue since the force process signal been fired.");
-                    }
-                    else
-                    {
-                        Trace.Verbose("Process timeline update queue since the time delay has met.");
-                    }
-                }
-
                 List<PendingTimelineRecord> pendingUpdates = new List<PendingTimelineRecord>();
-                lock (_timelineUpdateQueueLock)
+                foreach (var timeline in _allTimelines)
                 {
-                    foreach (var timeline in _allTimelines)
+                    ConcurrentQueue<TimelineRecord> recordQueue;
+                    if (_timelineUpdateQueue.TryGetValue(timeline, out recordQueue))
```

Comment:
Again, why would TryGet return false here? Consider adding a comment if this is a valid case.